bundles/org.openhab.binding.onewiregpio/src/main/java/org/openhab/binding/onewiregpio/internal/handler/OneWireGPIOHandler.java

```diff
@@ -84,21 +93,25 @@ public void initialize() {
      * When invalid parameter is found, default value is assigned.
      */
     private boolean checkConfiguration() {
-        Configuration configuration = getConfig();
-        gpioBusFile = (String) configuration.get(GPIO_BUS_FILE);
         if (StringUtils.isEmpty(gpioBusFile)) {
             logger.debug("GPIO_BUS_FILE not set. Please check configuration, and set proper path to w1_slave file.");
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
                     "The path to the w1_slave sensor data file is missing.");
             return false;
         }
 
-        refreshTime = ((Number) configuration.get(REFRESH_TIME)).intValue();
-        if (refreshTime.intValue() <= 0) {
-            logger.warn("Refresh time [{}] is not valid. Falling back to default value: {}.", refreshTime,
+        if (refreshTime <= 0) {
+            logger.debug("Refresh time [{}] is not valid. Falling back to default value: {}.", refreshTime,
                     DEFAULT_REFRESH_TIME);
             refreshTime = DEFAULT_REFRESH_TIME;
         }
+
+        if (precision < 0 || precision > MAX_PRECISION_VALUE) {
+            logger.debug("Precision value {} is outside allowed values [0 - 3]. Falling back to default value: {}.",
```

Comment:
Consider using the constant 3 as well here to keep the logging correct when the constant is changes in the future